lib/ansible/plugins/action/template.py

```diff
@@ -58,6 +58,7 @@ def run(self, tmp=None, task_vars=None):
         block_end_string = self._task.args.get('block_end_string', None)
         trim_blocks = boolean(self._task.args.get('trim_blocks', True), strict=False)
         lstrip_blocks = boolean(self._task.args.get('lstrip_blocks', False), strict=False)
+        output_encoding = self._task.args.get('output_encoding', 'utf-8') or 'utf-8'
```

Comment:
Is there a reason for the ```or``` here?

Reply:
Yes, I used a construct I've seen in the code for encoding fallback:

`test/units/mock/vault_helper.py`
```py
class TextVaultSecret(VaultSecret):
    '''A secret piece of text. ie, a password. Tracks text encoding.

    The text encoding of the text may not be the default text encoding so
    we keep track of the encoding so we encode it to the same bytes.'''

    def __init__(self, text, encoding=None, errors=None, _bytes=None):
        super(TextVaultSecret, self).__init__()
        self.text = text
        self.encoding = encoding or 'utf-8'
        self._bytes = _bytes
        self.errors = errors or 'strict'
```

`lib/ansible/parsing/vault/__init__.py`
```py
# TODO: mv these classes to a seperate file so we don't pollute vault with 'subprocess' etc
class FileVaultSecret(VaultSecret):
    def __init__(self, filename=None, encoding=None, loader=None):
        super(FileVaultSecret, self).__init__()
        self.filename = filename
        self.loader = loader

        self.encoding = encoding or 'utf8'

        # We could load from file here, but that is eventually a pain to test
        self._bytes = None
        self._text = None
```